src/lib/homestore_backend/hs_shard_manager.cpp

```diff
@@ -111,6 +111,16 @@ ShardManager::AsyncResult< ShardInfo > HSHomeObject::_create_shard(pg_id_t pg_ow
     auto new_shard_id = generate_new_shard_id(pg_owner);
     auto create_time = get_current_timestamp();
 
+    // select chunk for shard.
+    const auto p_chunkID = chunk_selector()->peek_top_chunk(pg_owner);
```

Comment:
refine 'peek_top_chunk' to 'pop_top_chunk'.

 Added handling for `CREATE_SHARD_MSG` type in `HSHomeObject::on_shard_message_rollback`. 
 
Since the leader made changes to the heap in `_create_shard` , the heap states of the leader and follower will be inconsistent during `on_shard_message_commit`. Therefore, modifications were made to HeapChunkSelector::select_specific_chunk and HeapChunkSelector::release_chunk.

Reply:
I think we need to add it to the `on_error`, this is the leader specific path that it cannot make a raft proposal.

Reply:
Could you please explain `on_error` in more detail? `pop_top_chunk` truely is an out-of-band operation by the leader, independent of the state machine. I'm not quite sure in what scenarios on_error is triggered and how it operates.

Reply:
I cannot list all possibility but the most common one I saw is leader swtich --- i.e if the node is no longer a leader, it cannot propose to raft and the on_error() callback will be called.

Reply:
done